src/seq.rs

```diff
@@ -474,6 +483,19 @@ pub struct Addr {
     pub port: i32,
 }
 
+impl FromStr for Addr {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let split: Vec<&str> = s.trim()
+                                .split(":")
+                                .collect();
+        let client = split[0].parse::<i32>()?;
```

Comment:
You don't need to collect into a Vec - just call `next()` on the iterator itself.

Also, it does not error out correctly in case you have too many or too few ':', i e, it should not panic on strings like '', '123', and '1:2:3'.

Reply:
Updated in the latest commit